internal/service/projects/main.go

```diff
@@ -1,22 +1,267 @@
+		}
+
+		var asset types.Asset
+		err = s.db.Where("id = ?", assetUUID).First(&asset).Error
+		if err != nil {
+			if err == gorm.ErrRecordNotFound {
+				return &openapi.ValidationError{ErrorMessage: fmt.Sprintf("Asset with ID [%v] does not exist", assetUUID)}, nil
+			}
+			return nil, types.NewErrFromGorm(err, "failed to fetch asset")
+		}
+
+		// Validate asset belongs to the study
+		if asset.StudyID != studyUUID {
+			return &openapi.ValidationError{ErrorMessage: fmt.Sprintf("Asset [%v] does not belong to the specified study", asset.Title)}, nil
+		}
+
+		// Validate asset tier is compatible with environment tier
+		if asset.Tier > environmentTier {
+			return &openapi.ValidationError{ErrorMessage: fmt.Sprintf("Asset [%v] has tier %d which is incompatible with environment (max tier %d)", asset.Title, asset.Tier, environmentTier)}, nil
+		}
+	}
+
+	return nil, nil
+}
+
+func (s *Service) createProjectTRERoleBindings(tx *gorm.DB, projectTREID uuid.UUID, members []openapi.ProjectTREMember) error {
+	for _, member := range members {
+		// Get user from database
+		user, err := s.users.PersistedUser(types.Username(member.Username))
+		if err != nil {
+			return types.NewErrServerError(fmt.Errorf("failed to get user %s: %w", member.Username, err))
+		}
+
+		// Create a role binding for each role assigned to this member
+		for _, role := range member.Roles {
+			roleBinding := types.ProjectTRERoleBinding{
+				ProjectTREID: projectTREID,
+				UserID:       user.ID,
+				Role:         types.ProjectTRERoleName(role),
+			}
+
+			if err := tx.Create(&roleBinding).Error; err != nil {
+				return types.NewErrFromGorm(err, fmt.Sprintf("failed to create role binding for user %s", member.Username))
+			}
+		}
+	}
+	return nil
+}
+
+func (s *Service) CreateProjectTRE(ctx context.Context, creator types.User, studyUUID uuid.UUID, projectTreData openapi.ProjectTRERequest) error {
+	// Get TRE environment
+	var treEnvironment types.Environment
+	err := s.db.Where("name = ?", "ARC Trusted Research Environment").First(&treEnvironment).Error
+	if err != nil {
+		return types.NewErrFromGorm(err, "failed to fetch TRE environment")
+	}
+
+	// Start a transaction
+	tx := s.db.Begin()
+	defer func() {
+		if r := recover(); r != nil {
+			tx.Rollback()
+		}
+	}()
+
+	// Create Project record
+	project := types.Project{
+		Name:           projectTreData.Name,
+		CreatorUserID:  creator.ID,
+		StudyID:        studyUUID,
+		ApprovalStatus: string(projectTreData.ApprovalStatus),
+	}
+
+	if err := tx.Create(&project).Error; err != nil {
+		tx.Rollback()
+		return types.NewErrFromGorm(err, "failed to create project")
+	}
+
+	// Create ProjectTRE record
+	projectTRE := types.ProjectTRE{
+		ProjectID:                     project.ID,
+		EnvironmentID:                 treEnvironment.ID,
+		EgressNumberRequiredApprovals: 1, // TODO: discuss with team if this should be configurable
+	}
```

Comment:
Not quite sure how to approach the `EgressNumberRequiredApprovals `. Any thoughts?